LibertyDSNP/provider-dashboard
# svelte store with local storage

closes #149 

## Changes Made:
- Added `storable` file that reads and writes from local storage to our desired stores.
- If the user has a stored network, connect to API using that network on load.
  - Because of this, we don't have to worry about connect to the API in any of the components, so I removed that code from `BecomeAProvider` and `LoginForm`.
  -   We could also simplify our functions in `polkadotApi.ts` to only use createApi().
- Made the state more reactive where possible for now.
- If viewing dashboard and the user has a stored network, fetchAccountsForNetwork, setting the accounts store reactivly.
- Swap out `afterUpdate()` or `onMount()` for `$:{}` where applicable so we can subscribe and update when changes happen, not only on load or infinite times with after update.
- Converted type URL to type string so that they could be parsed without problems when stored in local storage.
- Updated dropdowns to use .get() when setting the initial value so that it will be selected or display the placeholder (was blank).
- added some ids for testing
- updated e2e test.

## How to test
1. Login Flow:
- Login
- See expected dashboard
- Refresh and still see expected dashboard
- Click on logout button and be taken to login screen.
- Refresh, should still be logged out.
2. Become Provider Flow:
- Click on become a provider button
- Go to become provider page.
- Flow works as expected
- logged in and taken to dashboard on success
- Refresh and still see expected dashboard
3. Switch provider modal should function as expected.
4. Add key modal should function as expected.
5. Stake to provider should function as expected.
6. Logout should function as expected
7. Navigating to and from FAQ should not clear the data.
